bluemira/codes/params.py

```diff
@@ -159,3 +160,25 @@ def __setattr__(self, attr: str, value: Union[bool, str]):
             raise ValueError(f"{attr} must be a bool")
         else:
             super().__setattr__(attr, value)
+
+
+def make_mapped_default_parameter_frame(
+    defaults: dataclass, mappings: Dict[str, ParameterMapping], param_cls: Type[_PfT]
+) -> _PfT:
+    """
+    Create ParameterFrame with default values for external codes.
+
+    External codes are likely to have variables that are not changed often
+    therefore in some cases sane defaults are needed
+    """
+    new_param_dict = {}
+    for map_name, param_map in mappings.items():
+        if param_map.send:
+            new_param_dict[map_name] = {
+                "value": getattr(defaults, param_map.name),
+                "unit": param_map.unit,
+            }
+        else:
+            new_param_dict[map_name] = {"value": 0, "unit": param_map.unit}
```

Comment:
I dont really like this default but I couldn't think of a better option atm. Possibly at least have the same type